includes/class-wc-cart.php

```diff
@@ -660,18 +660,18 @@ public function get_cart_contents_count() {
 	 * Get weight of items in the cart.
 	 *
 	 * @since 2.5.0
-	 * @return int
+	 * @return float
 	 */
 	public function get_cart_contents_weight() {
-		$weight = 0;
+		$weight = 0.0;
 
 		foreach ( $this->get_cart() as $cart_item_key => $values ) {
 			if ( $values['data']->has_weight() ) {
 				$weight += (float) $values['data']->get_weight() * $values['quantity'];
 			}
 		}
 
-		return apply_filters( 'woocommerce_cart_contents_weight', $weight );
+		return (float) apply_filters( 'woocommerce_cart_contents_weight', $weight );
```

Comment:
@seostudio Thanks for the PR. This looks fine except for casting the filtered value to a float. This could break existing usage.

Reply:
@rrennick fixed